src/python/pants/backend/codegen/tasks/simple_codegen_task.py

```diff
@@ -13,11 +15,30 @@
 from pants.base.address import SyntheticAddress
 from pants.base.build_environment import get_buildroot
 from pants.base.exceptions import TaskError
+from pants.util.dirutil import safe_walk
+from pants.util.meta import AbstractClass
 
 
+logger = logging.getLogger(__name__)
+
 class SimpleCodegenTask(Task):
   """A base-class for code generation for a single target language."""
 
+  def __init__(self, *args, **kwargs):
+    super(SimpleCodegenTask, self).__init__(*args, **kwargs)
+    self._codegen_strategy = None
+
+  @classmethod
+  def register_options(cls, register):
+    super(SimpleCodegenTask, cls).register_options(register)
+    register('--strategy', choices=['global', 'isolated'], default='isolated',
+             help='Selects the compilation strategy to use. The "global" strategy uses a shared '
+                  'global directory for all generated code, and the "isolated" strategy uses '
+                  'per-target codegen directories.')
+    register('--allowempty', choices=['yes', 'no'], default='yes',
+             help='Whether code-generation should fail if a target specifies no sources, or just'
+                  'skip that target.')
```

Comment:
I found this necessary to add after testing in our /java repo, and finding that protoc failed on many things using the isolated strategy because we have proto targets that apparently don't have any sources.